packages/react/slot/src/Slot.tsx

```diff
@@ -22,7 +22,12 @@ const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
       if (child === slottable) {
         // because the new element will be the one rendered, we are only interested
         // in grabbing its children (`newElement.props.children`)
-        if (React.Children.count(newElement) > 1) return React.Children.only(null);
+        if (1 < React.Children.count(newElement)) {
+          throw new Error(
+            `\`asChild\` cannot be set to true when passing multiple ReactNodes as children.`
```

Comment:
This messaging here in `Slot` bothers me because this isn't where `asChild` is declared (that's in primitive).

Slot is a public utility and people who use it may not have an `asChild` or have a different prop.

I'll looking for a different approach.